framework/src/base/BlockRestrictable.C

```diff
@@ -27,6 +27,8 @@ InputParameters validParams<BlockRestrictable>()
   // Add the user-facing 'block' input parameter
   params.addParam<std::vector<SubdomainName> >("block", "The list of block ids (SubdomainID) that this object will be applied");
 
+  params.addPrivateParam<bool>("delay_initialization", false);
```

Comment:
Ok - if all MeshModifiers aren't always delaying BoundaryRestrictable initialization... that's a reason to have it as a parameter...

This is off-topic but my issue with the private parameters is that there's absolutely ZERO documentation that they exist.  How is someone making a new class supposed to even know that they have the option of delaying BoundaryRestrictable?